src/services/ImageRepository.h

```diff
@@ -20,37 +20,65 @@ enum class ImageType
 class ImageReference
 {
 public:
-    ImageReference() {}
-    ImageReference(ImageType nType, const std::string& sName)
+    ImageReference() noexcept = default;
+    explicit ImageReference(ImageType nType, const std::string& sName) noexcept
     {
         ChangeReference(nType, sName);
     }
 
-    ImageReference(const ImageReference& source)
-        : m_nType(source.m_nType), m_sName(source.m_sName)
+    ImageReference(const ImageReference& source) noexcept :
+        m_nType{ source.m_nType }, m_sName{ source.m_sName }
     {
         // don't copy m_hBitmap, it'll get initialized when GetHBitmap is called, which
         // will ensure the reference count is accurate.
     }
 
-    ~ImageReference();
-    
-    /// <summary>
-    /// Updates the referenced image.
-    /// </summary>
+    ImageReference& operator=(const ImageReference&) noexcept = delete;
+
+    ImageReference(ImageReference&& source) noexcept :
+        m_nType{source.m_nType},
+        m_sName{std::move(source.m_sName)},
+        m_hBitmap{ source.m_hBitmap }
+    {
+        source.m_nType = ImageType{};
+        if (source.m_hBitmap != nullptr)
+        {
+            DeleteBitmap(source.m_hBitmap);
+            source.m_hBitmap = nullptr;
+        }
+    }
+
+    ImageReference& operator=(ImageReference&& source) noexcept
+    {
+        m_nType   = source.m_nType;
+        m_sName   = std::move(source.m_sName);
+        m_hBitmap = source.m_hBitmap;
+
+        source.m_nType = ImageType{};
+        if (source.m_hBitmap != nullptr)
+        {
+            DeleteBitmap(source.m_hBitmap);
+            source.m_hBitmap = nullptr;
+        }
+
+        return *this;
+    }
```

Comment:
I needed to make sure `AchievementOverlay `wouldn't throw an exception during the move assign in `RA_Core.cpp`. From an end-user perspective nothing has changed, memory usage did go down a little bit though.

Images still show up.